src/App/Services/Tamara_Client.php

```diff
@@ -77,24 +77,25 @@ public function reinit_tamara_client( $api_token, $api_url = 'https://api.tamara
 	 * Create Tamara Checkout session, if failed, put errors to `wc_notices`
 	 *
 	 * @param $wc_order_id
+	 * @param $checkout_payment_type
+	 * @param $instalment_period
 	 *
 	 * @return array|bool
 	 * @throws \Exception
 	 */
-	public function process_payment( $wc_order_id ) {
+	public function process_payment( $wc_order_id, $checkout_payment_type, $instalment_period ) {
 		$wc_order = wc_get_order( $wc_order_id );
-		$instalment_period = 3;
-		$checkout_payment_type = 'PAY_BY_INSTALMENTS';
 		try {
 			$create_tamara_checkout_session_response = $this->build_checkout_session_request(
 				$wc_order,
 				$checkout_payment_type,
 				$instalment_period
 			);
+
 		} catch ( RequestDispatcherException $tamara_request_dispatcher_exception ) {
 			$error_message = General_Helper::convert_message( $tamara_request_dispatcher_exception->getMessage() );
 		} catch ( Exception $tamara_checkout_exception ) {
-			$error_message = $this->_t( 'Tamara Service unavailable! Please try again later.' ) . "<br />\n" . $this->_t( $tamara_checkout_exception->getMessage() );
+			$error_message = General_Helper::convert_message( 'Tamara Service unavailable! Please try again later.' ) . "<br />\n" . General_Helper::convert_message( $tamara_checkout_exception->getMessage() );
```

Comment:
Why do we need to use this ` General_Helper::convert_message`, this method is mainly for non-readable Tamara error code like `max_limit_reach`